app.py

```diff
@@ -174,21 +214,33 @@ def remove_gridoption_cellstyle(header_name: str, algorithm_name: str) -> None:
     logging.info(f"Assets : {assets}")
 
     with placeholder.container():
-        page_selected = st.selectbox(
-            "Which page of the following pdf contains the table you want to extract ?",
-            assets["pagefilter"]["selected_pages"],
-            index=None,
-            placeholder="Select a page number",
-        )
+        col1, col2 = st.columns([1, 1])
+        with col1:
+            st.markdown(
+                get_pdf_iframe(pdf_before_page_validation),
+                unsafe_allow_html=True,
+            )
 
-        st.markdown(get_pdf_iframe(pdf_before_page_validation), unsafe_allow_html=True)
+        with col2, st.form(key="selected_pages_form"):
+            number_pages = (
+                len(PdfReader(pdf_before_page_validation).pages) + 1
+            )  # Make the index start to 1
+            pages_selected = st.multiselect(
+                "Which page of the following pdf contains the table you want to extract ?",
+                list(range(1, number_pages)),
+                placeholder="Select a page number",
+            )
+            submitted = st.form_submit_button(
+                label="Validate your selected pages",
+                on_click=set_validate,
+                args=("validate_selected_pages",),
+            )
 
-        if page_selected == "None":
+        if pages_selected == ["None"]:  # TODO : check this behavior
             number_pages = len(pdfreader.pages)
```

Comment:
```suggestion
        if pages_selected == []:  # TODO : check this behavior
```
Si tu veux un comportement qui bloque l'utilisateur à cette étape tant qu'il n'a pas fait de choix

Reply:
Il doit aussi y avoir moyen de faire la vérif' sur le callback du bouton ou avec un 
```
while pages_selected == []:
 st.multiselect(...)
```
qu'est-ce que tu en penses ?

Reply:
alors en fait je le suis rendu compte qu'en terme d'UX c'était un enfer ce "None" ajouté manuellement dans la liste du coup dans la prochaine PR je l'ai supprimé et j'ai ajouté un task dans la todo d'ajouter un bouton spécifique pour gérer ce comportement (ie le tableau pas détécté par l'arbre de décision